src/services/auth/auth.service.ts

```diff
@@ -130,17 +138,43 @@ export class AuthService {
     return await this.sessionRepository.findOneBySub(this.generateDidSub(did));
   }
 
-  async createEmailAndPasswordUser(
+  async #createEmailAndPasswordUser(
     email: string,
     password: string,
     user_id: string,
   ): Promise<string> {
-    await this.legacyUserRepository.createNewEmailUser({ email, user_id });
-    return await this.legacyUserAccountRepository.createNewEmailAndPasswordUser({
-      email,
-      password,
-      username: user_id,
-    });
+    try {
+      await this.legacyUserRepository.createNewEmailUser({ email, user_id });
+      return await this.legacyUserAccountRepository.createNewEmailAndPasswordUser({
```

Comment:
I'm not sure if you are doing this already, but the account shouldn't be active until the registration is completed

Reply:
I've made some changes to support this and its raised a question further down the pr